src/feature_filter/feature_filter.test.ts

```diff
@@ -329,9 +329,81 @@ function legacyFilterTests(createFilterExpr) {
     });
 
     test('==, $type', () => {
-        const f = createFilterExpr(['==', '$type', 'LineString']).filter;
-        expect(f({zoom: 0}, {type: 1})).toBe(false);
-        expect(f({zoom: 0}, {type: 2})).toBe(true);
+        const fPoint = createFilterExpr(['==', '$type', 'Point']).filter;
```

Comment:
Can this test be split into a few different tests to indicate what is being tested?